src/Tribe/Repositories/Traits/Post_Tickets.php

```diff
@@ -0,0 +1,288 @@
+					AND (
+						{$prefix}_ticket_cost.meta_key REGEXP %s
+						OR {$prefix}_ticket_cost.meta_id IS NULL
+					)
+			)", $price_regexp ) );
+
+		$prepared_value = is_array( $value ) ? $repo->prepare_interval( $value, '%d', $operator ) : $wpdb->prepare( '%d', $value );
+
+		// Default the cost to `0` if not set to make RSVP tickets show as "free" tickets, with a cost of 0.
+		$repo->where_clause( "IFNULL( {$prefix}_ticket_cost.meta_value, 0 ) {$operator} {$prepared_value}" );
+
+		if ( null !== $symbol ) {
+			$this->filter_by_cost_currency_symbol( $symbol );
+		}
+	}
+
+	/**
+	 * Filters events that have a ticket with a specific cost currency symbol.
+	 *
+	 * Events with a cost of `0` but a currency symbol set will be fetched when fetching
+	 * by their symbols; RSVP tickets have no symbol and will never match any filtering
+	 * by currency symbol.
+	 * Filtering by currency symbol, when done in the context of Event Tickets, really means
+	 * filtering events by tickets that come from providers with a specific currency ISO code.
+	 * As an example filtering by "USD" when Tribe Commerce tickets use the "EUR" code and
+	 * WooCommerce tickets use the "USD" code means "only fetch events that have WooCommerce
+	 * tickets".
+	 *
+	 * @since TBD
+	 *
+	 * @param string|array $symbol One or more currency symbols or currency ISO codes. E.g.
+	 *                             "$" and "USD".
+	 *
+	 * @throws Tribe__Repository__Void_Query_Exception If no provider uses the specified currency symbol
+	 *                                                 or ISO code.
+	 */
+	public function filter_by_cost_currency_symbol( $symbol ) {
+		$repo = $this;
+
+		// If the repo is decorated, use that.
+		if ( ! empty( $repo ) ) {
+			$repo = $this->decorated;
+		}
+
+		/** @var Tribe__Tickets__Commerce__Currency $currency */
+		$currency      = tribe( 'tickets.commerce.currency' );
+		$symbols       = (array) $symbol;
+		$request_codes = [];
+
+		/*
+		 * Transform the request symbols into ISO codes; due to its ambiguous nature a
+		 * symbol might match 0+ ISO codes.
+		 */
+		foreach ( $symbols as $request_symbol ) {
+			$request_codes[] = (array) $currency->get_symbol_codes( $request_symbol );
+		}
+		$request_codes = array_unique( call_user_func_array( 'array_merge', $request_codes ) );
+
+		if ( empty( $request_codes ) ) {
+			$reason = 'The specified currency symbol or ISO code is not supported.';
+			throw Tribe__Repository__Void_Query_Exception::because_the_query_would_yield_no_results( $reason );
+		}
+
+		// Compile a list of ticket providers that are active and use one of the requested ISO codes.
+		$request_providers = [];
+
+		if ( array_intersect( $request_codes, (array) $currency->get_currency_code() ) ) {
+			$request_providers[] = 'tpp';
+		}
+
+		$providers = [
+			'eddticket' => 'Tribe__Tickets_Plus__Commerce__EDD__Main',
+			'wooticket' => 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main',
```

Comment:
use class constant?

Reply:
This is just brought over from existing ET code (just put into a trait so it can be used for both Events and the new Post repository. The specific wooticket/eddticket part is passed into the SQL regex below, so it could be added into a better abstraction at some point in the future. The code was passing automated tests and I didn't want to mess with it too much further because this ticket went over on effort.